src/components/ShareDialog.jsx

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { makeStyles } from '@material-ui/core/styles';
+import {
+  Dialog,
+  DialogContent,
+  DialogContentText,
+  DialogTitle,
+} from '@material-ui/core';
+import {
+  FacebookShareButton,
+  FacebookIcon,
+  EmailShareButton,
+  EmailIcon,
+  LinkedinShareButton,
+  LinkedinIcon,
+  RedditShareButton,
+  RedditIcon,
+  TwitterShareButton,
+  TwitterIcon,
+} from 'react-share';
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    background: theme.palette.background.secondary,
+    color: 'white',
+    fontSize: '15px',
+    textAlign: 'center',
+  },
+  shareIcon: {
+    maringRight: '15px',
+  },
+}));
+
+export default function ShareDialog(props) {
+  const classes = useStyles();
+  const { open, onClose, link, title } = props;
+
+  return (
+    <Dialog open={open} onClose={onClose}>
+      <DialogTitle className={classes.root}>Share</DialogTitle>
+      <DialogContent className={classes.root}>
+        <DialogContentText className={classes.root}>
+          Currently Sharing: {title}
+        </DialogContentText>
+        <EmailShareButton url={link}>
+          <EmailIcon size={55} round />
```

Comment:
The icons ought to similarly also be based on styles if possible. I'm not sure how size works here but ideally it'd be a multiple of 8 (I think 64 is the default if you don't specify size).

Reply:
Yeah but since `theme` is not imported in this area im not sure if using theme.spacing(N) would work? Default size is 64 and I removed the size value for the component but we can resize if its too big?

Reply:
The default looks fine to me. There is actually a way to use the theme by using a ThemeProvider but I think the default size is fine.

Reply:
Sure sounds good, we will leave them as default